plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -600,14 +599,16 @@ function OPDSBrowser:showDownloads(item)
                 {
                     text = _("Page stream") .. "\u{2B0C}", -- append LEFT RIGHT BLACK ARROW
                     callback = function()
-                        self:streamPages(item, acquisition.href, acquisition.count)
+                        OPDSPSE:storeCatalogCredentials(self.root_catalog_username, self.root_catalog_password)
+                        OPDSPSE:streamPages(item, acquisition.href, acquisition.count, false)
                         UIManager:close(self.download_dialog)
                     end,
                 },
                 {
                     text = _("Stream from page") .. "\u{2B0C}", -- append LEFT RIGHT BLACK ARROW
                     callback = function()
-                        self:streamPages(item, acquisition.href, acquisition.count, true)
+                        OPDSPSE:storeCatalogCredentials(self.root_catalog_username, self.root_catalog_password)
+                        OPDSPSE:streamPages(item, acquisition.href, acquisition.count, true)
```

Comment:
Maybe just pass username and password as arguments to `streamPages`?
(again, item doesn't have username and password fields)

Reply:
Ok. I'll do that.